src/ScheduledTask/Webhook/CancelOrRefundWebhookHandler.php

```diff
@@ -0,0 +1,137 @@
+<?php declare(strict_types=1);
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2022 Adyen N.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <shopware@adyen.com>
+ */
+
+namespace Adyen\Shopware\ScheduledTask\Webhook;
+
+use Adyen\Shopware\Entity\Notification\NotificationEntity;
+use Adyen\Shopware\Entity\Refund\RefundEntity;
+use Adyen\Shopware\Service\RefundService;
+use Adyen\Util\Currency;
+use Adyen\Webhook\PaymentStates;
+use Shopware\Core\Checkout\Order\Aggregate\OrderTransaction\OrderTransactionEntity;
+use Shopware\Core\Checkout\Order\Aggregate\OrderTransaction\OrderTransactionStates;
+use Shopware\Core\Framework\Context;
+
+class CancelOrRefundWebhookHandler implements WebhookHandlerInterface
+{
+    /**
+     * @var RefundService
+     */
+    private $refundService;
+
+    /**
+     * @param RefundService $refundService
+     * @return void
+     */
+    public function __construct(RefundService $refundService)
+    {
+        $this->refundService = $refundService;
+    }
+
+    /**
+     * @param OrderTransactionEntity $orderTransactionEntity
+     * @param NotificationEntity $notificationEntity
+     * @param string $state
+     * @param string $currentTransactionState
+     * @param Context $context
+     * @return OrderTransactionEntity
+     * @throws \Adyen\AdyenException
+     */
+    public function handleWebhook(
+        OrderTransactionEntity $orderTransactionEntity,
+        NotificationEntity $notificationEntity,
+        string $state,
+        string $currentTransactionState,
+        Context $context
+    ): OrderTransactionEntity {
+        if ($notificationEntity->isSuccess() && $state !== $currentTransactionState) {
+            // Process refund notification
+            if ($state === PaymentStates::STATE_REFUNDED) {
+                $this->handleSuccessfulRefund($orderTransactionEntity, $notificationEntity, $context);
```

Comment:
Can we update the `handleSuccessfulNotification` and `handleFailedNotification` functions to return the OrderTransactionEntity, so that the `handleWebhook` returns the updated OrderTransactionEntity, instead of the same one passed to this function?

Reply:
Resolved since Shopware 6 does not return the updated orderTransactionEntity